src/nvim/edit.c

```diff
@@ -2485,6 +2485,64 @@ void set_completion(colnr_T startcol, list_T *list)
 }
 
 
+// complete_mode() implementation.
+const char *ins_compl_mode(void)
```

Comment:
`const char *` would imply caller shouldn't free. But we could keep `const char*` here and remove all `xstrdup` and only have one `xstrdup` call in `f_complete_mode` instead.

Reply:
OK.  I have fixed it.